libqtile/core/manager.py

```diff
@@ -330,6 +330,15 @@ def _process_screens(self) -> None:
             scr._configure(self, i, x, y, w, h, grp)
             self.screens.append(scr)
 
+        self.x_screen_dimensions: Dict[str, int] = {'x': 0, 'y': 0}
+        for screen in self.screens:
+            if screen.x is not None is not screen.width and \
+                    screen.y is not None is not screen.height:
+                self.x_screen_dimensions['x'] = max(self.x_screen_dimensions.get('x') or 0,
+                                                    screen.x + screen.width)
+                self.x_screen_dimensions['y'] = max(self.x_screen_dimensions.get('y') or 0,
+                                                    screen.y + screen.height)
```

Comment:
This calculates the X screen to be the smallest rectangle that fits all of the enabled outputs inside it. Is that always the case?

Also perhaps you can ask the X server for this information direction... try `qtile.root.get_geometry()`.

Reply:
Another small comment - could you change "x" and "y" to width and height, if that is what they represent? This would keep it consistent with other uses of these word in the code and avoid confusion

Reply:
If we only need this information ready when `add_strut` is called, maybe it can be move into that methods local scope

Reply:
Using root.get_geometry() seems to be the smarter way, I just did not know it existed. I moved it into add_strut(), otherwise I'd have preferred it to remain in _process_screens() to not reiterate again and again.